feat: add Docker containerization with health endpoints

MaksymChukhrai opened the pull request:
## Changes
- Docker multi-stage build (dev + prod)
- docker-compose with MySQL, Adminer
- Health endpoints: `/`, `/health`
- Health module with API info
- Fixed JWT `@Public()` decorator

## Details
See [docs/DOCKER.md](docs/DOCKER.md)

## Testing
```bash
docker-compose up -d
curl http://localhost:3000/health
```

MaksymChukhrai committed:
feat: add Docker containerization with health endpoints

MaksymChukhrai committed:
refactor: remove some comments

lyudad commented on src/modules/auth/auth.controller.ts:
```diff
@@ -1,188 +1,74 @@
-    @Body() refreshTokenDto: RefreshTokenDto,
-  ): Promise<{ accessToken: string; refreshToken: string }> {
-    return await this.authService.refresh(refreshTokenDto.refreshToken);
+  async refresh(@Body() dto: RefreshTokenDto) {
```
1.1.1

MaksymChukhrai replied:
Fixed - added explicit return type: Promise<AuthResponseDto>

lyudad commented on src/modules/auth/auth.controller.ts:
```diff
@@ -1,188 +1,74 @@
-    @CurrentUser() user: UserEntity,
-  ): Promise<{ message: string }> {
-    return await this.authService.logout(user.id);
+  async logout(@Req() req: Request) {
```
1.1.1

MaksymChukhrai replied:
Fixed - replaced with @CurrentUser() decorator, removed Express import.

lyudad commented on src/modules/auth/auth.controller.ts:
```diff
@@ -1,188 +1,74 @@
+  // @Get('facebook')
+  // async facebookAuth() {
+  //   // OAuth redirect
+  // }
```
delete. no one will use this code

MaksymChukhrai replied:
Removed. Was keeping as placeholder for Sprint 1 Week 2 OAuth tasks, but understand it creates noise.

lyudad commented on src/modules/auth/auth.controller.ts:
```diff
@@ -1,188 +1,74 @@
+  // @Public()
+  // @Post('magic-link')
+  // async magicLink(@Body() dto: MagicLinkDto) {
+  //   // Send magic link email
```
delete noone will use this code

MaksymChukhrai replied:
Removed in latest commit.

lyudad commented on src/modules/auth/auth.module.ts:
```diff
@@ -1,27 +1,38 @@
 import { JwtModule } from '@nestjs/jwt';
 import { PassportModule } from '@nestjs/passport';
-import { TypeOrmModule } from '@nestjs/typeorm';
+// import { TypeOrmModule } from '@nestjs/typeorm';
```
1.1.10

MaksymChukhrai replied:
Removed commented import. Will add back when DB migration is ready.

MaksymChukhrai replied:
Removed duplicate comment. TypeORM integration pending DB setup.

lyudad commented on src/modules/auth/auth.module.ts:
```diff
@@ -1,27 +1,38 @@
 import { AuthController } from './auth.controller';
 import { AuthService } from './auth.service';
-import { UserEntity } from './user.entity';
+// import { User } from './user.entity';
```
1.1.10

MaksymChukhrai replied:
Removed duplicate comment.

lyudad commented on src/modules/auth/auth.module.ts:
```diff
@@ -1,27 +1,38 @@
         secret: configService.get<string>('JWT_SECRET'),
         signOptions: {
-          expiresIn: configService.get<string>('JWT_EXPIRES_IN'),
+          expiresIn: configService.get<string>('JWT_EXPIRATION'), // ✅ Исправлено
```
why you dont check what you add?

MaksymChukhrai replied:
My mistake - added comment during debugging, removed now. Will check all changes before commit.

MaksymChukhrai committed:
fix: add explicit return types and remove unnecessary comments per PR feedback

- Added Promise return types for all auth endpoints
- Removed Express Request import, using @CurrentUser decorator
- Cleaned up commented code placeholders
- Fixed UserMockRepository to return Promises correctly
- Updated DTOs to match User entity structure

MaksymChukhrai commented:
## ✅ Fixed PR feedback issues

### Changes made:
- Added explicit return types for all controller methods (`Promise<AuthResponseDto>`, `Promise<UserResponseDto>`)
- Removed `import { Request } from 'express'` - now using `@CurrentUser()` decorator
- Removed commented OAuth placeholders (will implement in separate PR for Sprint 1 Week 2)
- Fixed `UserMockRepository` to properly return Promises
- Updated `UserResponseDto` to match current `User` entity structure

### Checklist applied:
- ✅ 1.1.1 types for input and output params
- ✅ 1.1.10 delete commented code
- ✅ All ESLint rules passing
- ✅ Build successful

### Note on TypeORM comments:
TypeORM imports remain commented temporarily until DB migration is ready (tracked separately by backend team). Will uncomment once real entities are connected.

Ready for re-review 🙏

MaksymChukhrai committed:
Merge develop into feature/implement-docker-be

Resolved merge conflicts:
- .env.example: Combined Docker + OAuth + AI variables
- src/app.module.ts: Added HealthModule (Docker) + AnalysisModule + UploadModule (AI)
- Auth module: Integrated Magic Link + OAuth infrastructure from develop
- Auth strategies: Updated to use TypeORM repositories instead of mocks
- Auth DTOs: Updated UserResponseDto, AuthResponseDto with new fields
- Removed login.dto.ts (replaced by Magic Link authentication)

Features integrated:
- Docker containerization (from feature/implement-docker-be)
- Magic Link passwordless authentication (from develop)
- Google Gemini AI blood test analysis (from develop)
- Upload module for test results (from develop)
- TypeORM with MySQL/SQLite support (from develop)

All auth code migrated to use UserEntity instead of User mock

MaksymChukhrai committed:
fix: Use SQLite for development, MySQL for production

- Added DB_TYPE environment variable check
- SQLite used when DB_TYPE=better-sqlite3
- MySQL used for production deployment
- Migration to MySQL will be in separate PR

MaksymChukhrai committed:
chore: Add better-sqlite3 dependency

- Installed better-sqlite3 for SQLite support in development
- Required for TypeORM SQLite driver

MaksymChukhrai committed:
chore: Add SQLite database files to .gitignore
